app/models/library.rb

```diff
@@ -0,0 +1,10 @@
+# typed: strict
+class Library < ApplicationRecord
+  belongs_to :user
+  has_many :styles
+  has_many :components
+
+  validates :description, presence: true
+  validates :user_id, presence: true
+end
```

Comment:
Consider adding a dependent: :destroy option to the has_many :components association to ensure that associated components are deleted when a library is deleted. This helps maintain data integrity.